strawberryfields/apps/subgraph.py

```diff
@@ -222,21 +260,16 @@ def resize(subgraph: list, graph: nx.Graph, min_size: int, max_size: int) -> dic
         graph (nx.Graph): the input graph
         min_size (int): minimum size for subgraph to be resized to
         max_size (int): maximum size for subgraph to be resized to
+        node_select (str, list or array): method of settling ties when more than one node of
+            equal degree can be added/removed. Can be ``"uniform"`` (default), or a numpy array or
+            list.
 
     Returns:
         dict[int, list[int]]: a dictionary of different sizes with corresponding subgraph
     """
     nodes = graph.nodes()
     subgraph = set(subgraph)
-
-    if not subgraph.issubset(nodes):
-        raise ValueError("Input is not a valid subgraph")
-    if min_size < 1:
-        raise ValueError("min_size must be at least 1")
-    if max_size >= len(nodes):
-        raise ValueError("max_size must be less than number of nodes in graph")
-    if max_size < min_size:
-        raise ValueError("max_size must not be less than min_size")
+    node_select, w = _validate_inputs(subgraph, graph, min_size, max_size, node_select)
```

Comment:
_validate_inputs gets defined after it is used here. Is that good coding practice?

Reply:
Since Python does not use name checking at 'compile' time, only when the defined function/method is executed, this is typically quite idiomatic in Python. You see it quite a lot in class methods as well.

Reply:
Thanks @josh146